### PR TITLE
improved initialization for large towns

### PR DESCRIPTION
**Pull request by mjanczykowski**
#32 All actors resolve their neighbors' actorRefs from paths before starting simulation.
Using resolveOne to check if actors are alive.

### CODE REVIEW

File: common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala [hunk 1]
```diff
@@ -1,34 +1,99 @@
...
+      val horizontalRoadsData = startSimulationMessage.horizontalRoadsDefs.map(r => AreaRoadData(r.roadId, r.direction, getActorRef(r.outgoingActorPath), getActorRef(r.prevAreaActorPath)))
+
+      val area = new Area(verticalRoadsData, horizontalRoadsData, config)
       // sending initial available road space info
       sendAvailableRoadSpaceInfo(area)
       self ! ReadyForComputation(0)
       unstashAll
-      goto(Simulating) using AreaData(area, None, x, y, supervisor, 0)
+      goto(Simulating) using AreaData(area, None, startSimulationMessage.x, startSimulationMessage.y, supervisor, 0)
+    case Event(msg@ResolvedActor(_, _), data@ResolvingActorsData(_, _, _, _)) =>
```
Comment (salceson) [inline common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala]:
```scala
case Event(msg: ResolvedActor, data: ResolvingActorsData) =>  ...
```

File: common/src/main/scala/miss/trafficsimulation/roads/Road.scala [hunk 2]
```diff
@@ -179,7 +179,7 @@ class RoadSegment(val roadId: RoadId,
     }
 
     //switch lanes not supported on initial part of firstRoadSegment
-    if(1==1) {
+    if (1 == 1) {
```
Comment (salceson) [inline common/src/main/scala/miss/trafficsimulation/roads/Road.scala:182]: I still don't like this, but let's call it not the scope of this PR.

File: common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala [hunk 3]
```diff
@@ -124,28 +189,32 @@ class AreaActor(config: Config) extends FSM[State, Data] with Stash {
...
   }
+
+  log.debug(s"Actor ${self.path} initialized")
 }
 
 object AreaActor {
 
   def props(config: Config): Props = Props(classOf[AreaActor], config)
 
+  case class AreaRoadDefinition(roadId: RoadId, direction: RoadDirection, outgoingActorPath: ActorPath, prevAreaActorPath: ActorPath) extends SerializableMessage
```
Comment (salceson) [inline common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala]: This line is a bit too long.

File: common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala [hunk 4]
```diff
@@ -1,34 +1,99 @@
...
+
+    resolveActorWithRetries(resolveActorRetries)
+  }
+
   when(Initialized) {
-    case Event(StartSimulation(verticalRoadsDefs, horizontalRoadsDefs, x, y), EmptyData) =>
+    case Event(msg@StartSimulation(verticalRoadsDefs, horizontalRoadsDefs, x, y), EmptyData) =>
+      val actorsToResolve = verticalRoadsDefs.flatMap(vrd => Set(vrd.prevAreaActorPath, vrd.outgoingActorPath)).toSet ++ horizontalRoadsDefs.flatMap(hrd => Set(hrd.prevAreaActorPath, hrd.outgoingActorPath)).toSet
+      resolveActor(actorsToResolve.head)
       val supervisor = sender()
```
Comment (salceson) [inline common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala]: IMO this should be the first line of handling this event.

**Review by salceson — CHANGES_REQUESTED** [review]
Some small changes, please.

File: common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala (hunk 4 above)
Comment (mjanczykowski) [reply common/src/main/scala/miss/trafficsimulation/actors/AreaActor.scala]: "val supervisor = sender()"?

**Review by salceson — APPROVED** [review]

**Pull request merged**